src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,168 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public MyLinkedList() {
+
+    }
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            Node<T> node = new Node<>(null, value, null);
+            head = node;
+            tail = node;
+            size++;
+            return;
+        }
+
+        Node<T> prev = tail;
+        Node<T> node = new Node<>(prev, value, null);
+        tail = node;
+        prev.next = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size && index >= 0) {
+            add(value);
+            return;
+        }
+        indexChecking(index);
+
+        Node<T> nodeByIndex = getNodeByIndex(index);
```

Comment:
This is code repeating. You always check index before getting node by index. What you should do is move the check logic INSIDE the get method.
![image](https://github.com/mate-academy/jv-linked-list/assets/20326858/c5b0f6b5-9440-4ad3-9e0f-1da4bf40d184)